tests/common-lib/index.js

```diff
@@ -0,0 +1,261 @@
+      .split('\n')
+      .filter((res) => res)
+  )
+
+  return resources
+}
+
+/**
+ * Return an imported cluster attached to the current hub cluster environment.
+ * @returns `targetCluster` The imported cluster.
+ */
+function getTargetManagedCluster() {
+  var targetCluster
+
+  try {
+    // Fetch imported clusters attached to the hub cluster.
+    var managedClusters = execSync(
+      'oc get managedclusters -o custom-columns=NAME:.metadata.name --no-headers'
+    )
+      .toString()
+      .split('\n').filter((cluster) => cluster)
+
+    console.info('Found the following clusters:', managedClusters)
+
+    // Check to see if the managed cluster name exist wihin the environment.
+    if (process.env.OPTIONS_MANAGED_CLUSTER_NAME) {
+      console.info(
+        `Checking for the exported managed cluster: ${process.env.OPTIONS_MANAGED_CLUSTER_NAME} within the returned list.`
+      )
+
+      if (
+        managedClusters.find((c) =>
+          c.includes(process.env.OPTIONS_MANAGED_CLUSTER_NAME)
+        )
+      ) {
+        targetCluster = process.env.OPTIONS_MANAGED_CLUSTER_NAME
+        return targetCluster
+      }
+
+      console.warn(
+        `The targeted managed cluster: ${process.env.OPTIONS_MANAGED_CLUSTER_NAME} was not found in the list: ${managedClusters}`
+      )
+    }
+
+    if (
+      managedClusters.length === 1 &&
+      managedClusters.find((c) => c.includes('local-cluster'))
+    ) {
+      console.info(
+        `Managed cluster list only contains one managed cluster: ${managedClusters}. Proceeding to test only the local-cluster.`
+      )
+      return targetCluster
+    } else {
+      // In the canary tests, we only need to focus on the import-xxxx managed cluster.
+      targetCluster = managedClusters.find(
+        (c) =>
+          c.startsWith('canary-') ||
+          c.includes('canary') ||
+          c.startsWith('import-')
+      )
+    }
+
+    if (targetCluster === undefined) {
+      targetCluster = managedClusters.find((c) => !c.includes('local-cluster'))
+    }
+
+    console.info(`Preparing to test with managed cluster: ${targetCluster}`)
+    return targetCluster
+  } catch (err) {
+    console.warn(
+      err,
+      'Error getting managedclusters. Proceeding with the local cluster.'
+    )
+    return
+  }
+}
+
+/**
+ * Filter the array to contain only non-empty values.
+ * @param {Array} array The array to filter the values from.
+ * @returns `array` The filtered array.
+ */
+ function removeEmptyEntries(array) {
+  return array.filter((val) => val.replace(/\s+/g, ' '))
+}
+
+/**
+ * 
+ * @param {*} kind 
+ * @param {*} resourceList 
+ * @param {*} requiredList 
+ * @returns 
+ */
+function shouldUseAPIGroup(kind, resourceList, requiredList = []) {
+  // Set useAPIGroup to false by default.
+  var useAPIGroup = false
+
+  const _ = resourceList.filter((res) => res.kind === kind)
+  if (_.length > 1 || requiredList.includes(kind)) useAPIGroup = true
+
+  return useAPIGroup
+}
```

Comment:
You could simplify like this.
```suggestion
function shouldUseAPIGroup(kind, resourceList, requiredList = []) {
  const _ = resourceList.filter((res) => res.kind === kind)
  return _.length > 1 || requiredList.includes(kind)
}
```